errors will be here{
  message: {
    getTrelloBoardError: FetchError: request to https://api.trello.com/1/organizations/ttpbetalocal/boards?fields=id,name&key=0d05ca858d1bc7893a91aef2712e7568&token=2d8a9892820b0ddb4d67225b6ab7f4130ccd31fffc6860c3f743e3bf6147af68 failed, reason: getaddrinfo EAI_AGAIN api.trello.com
        at ClientRequest.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/node-fetch/lib/index.js:1505:11)
        at ClientRequest.emit (node:events:527:28)
        at ClientRequest.emit (node:domain:475:12)
        at TLSSocket.socketErrorListener (node:_http_client:454:9)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at emitErrorNT (node:internal/streams/destroy:157:8)
        at emitErrorCloseNT (node:internal/streams/destroy:122:3)
        at processTicksAndRejections (node:internal/process/task_queues:83:21) {
      type: 'system',
      errno: 'EAI_AGAIN',
      code: 'EAI_AGAIN'
    }
  },
  level: 'error',
  timestamp: '2023-05-31 11:06:37'
}
{
  level: 'error',
  timestamp: '2023-05-31 11:06:37',
  message: "Cannot read properties of undefined (reading 'includes')",
  stack: "TypeError: Cannot read properties of undefined (reading 'includes')\n" +
    '    at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:102:17\n' +
    '    at Array.filter (<anonymous>)\n' +
    '    at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:101:35\n' +
    '    at Generator.next (<anonymous>)\n' +
    '    at fulfilled (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:5:58)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)'
}
{
  message: {
    getTrelloBoardError: FetchError: request to https://api.trello.com/1/organizations/ttpbetalocal/boards?fields=id,name&key=0d05ca858d1bc7893a91aef2712e7568&token=2d8a9892820b0ddb4d67225b6ab7f4130ccd31fffc6860c3f743e3bf6147af68 failed, reason: getaddrinfo EAI_AGAIN api.trello.com
        at ClientRequest.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/node-fetch/lib/index.js:1505:11)
        at ClientRequest.emit (node:events:527:28)
        at ClientRequest.emit (node:domain:475:12)
        at TLSSocket.socketErrorListener (node:_http_client:454:9)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at emitErrorNT (node:internal/streams/destroy:157:8)
        at emitErrorCloseNT (node:internal/streams/destroy:122:3)
        at processTicksAndRejections (node:internal/process/task_queues:83:21) {
      type: 'system',
      errno: 'EAI_AGAIN',
      code: 'EAI_AGAIN'
    }
  },
  level: 'error',
  timestamp: '2023-05-31 11:06:37'
}
{
  message: {
    error: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
        at Function.all (<anonymous>)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:322:28
        at Generator.next (<anonymous>)
        at fulfilled (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
  },
  level: 'error',
  timestamp: '2023-05-31 11:06:37'
}
{
  message: {
    getTrelloBoardError: FetchError: request to https://api.trello.com/1/organizations/ttpbetalocal/boards?fields=id,name&key=0d05ca858d1bc7893a91aef2712e7568&token=2d8a9892820b0ddb4d67225b6ab7f4130ccd31fffc6860c3f743e3bf6147af68 failed, reason: getaddrinfo EAI_AGAIN api.trello.com
        at ClientRequest.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/node-fetch/lib/index.js:1505:11)
        at ClientRequest.emit (node:events:527:28)
        at ClientRequest.emit (node:domain:475:12)
        at TLSSocket.socketErrorListener (node:_http_client:454:9)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at emitErrorNT (node:internal/streams/destroy:157:8)
        at emitErrorCloseNT (node:internal/streams/destroy:122:3)
        at processTicksAndRejections (node:internal/process/task_queues:83:21) {
      type: 'system',
      errno: 'EAI_AGAIN',
      code: 'EAI_AGAIN'
    }
  },
  level: 'error',
  timestamp: '2023-05-31 11:07:26'
}
{
  level: 'error',
  timestamp: '2023-05-31 11:07:26',
  message: "Cannot read properties of undefined (reading 'includes')",
  stack: "TypeError: Cannot read properties of undefined (reading 'includes')\n" +
    '    at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:107:28\n' +
    '    at Array.filter (<anonymous>)\n' +
    '    at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:106:41\n' +
    '    at Generator.next (<anonymous>)\n' +
    '    at fulfilled (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:5:58)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)'
}
{
  message: {
    getTrelloBoardError: FetchError: request to https://api.trello.com/1/organizations/ttpbetalocal/boards?fields=id,name&key=0d05ca858d1bc7893a91aef2712e7568&token=2d8a9892820b0ddb4d67225b6ab7f4130ccd31fffc6860c3f743e3bf6147af68 failed, reason: getaddrinfo EAI_AGAIN api.trello.com
        at ClientRequest.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/node-fetch/lib/index.js:1505:11)
        at ClientRequest.emit (node:events:527:28)
        at ClientRequest.emit (node:domain:475:12)
        at TLSSocket.socketErrorListener (node:_http_client:454:9)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at emitErrorNT (node:internal/streams/destroy:157:8)
        at emitErrorCloseNT (node:internal/streams/destroy:122:3)
        at processTicksAndRejections (node:internal/process/task_queues:83:21) {
      type: 'system',
      errno: 'EAI_AGAIN',
      code: 'EAI_AGAIN'
    }
  },
  level: 'error',
  timestamp: '2023-05-31 11:07:26'
}
{
  message: {
    error: TypeError: undefined is not iterable (cannot read property Symbol(Symbol.iterator))
        at Function.all (<anonymous>)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:322:28
        at Generator.next (<anonymous>)
        at fulfilled (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/startup/db/dbConnect.ts:5:58)
        at processTicksAndRejections (node:internal/process/task_queues:96:5)
  },
  level: 'error',
  timestamp: '2023-05-31 11:07:26'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166871,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166876,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166878,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:01:41'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166871,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166876,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166878,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:01:41'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166871,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166876,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166878,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:01:41'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166871,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166876,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166878,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:01:41'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166871,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166876,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166878,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:01:41'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166871,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166876,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100166878,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:01:41'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197060,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197062,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197066,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:02:11'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197060,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197062,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197066,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:02:11'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197060,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197062,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197066,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:02:11'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197060,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-00.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-00.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197062,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-01.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-01.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 100197066,
            lastWriteDate: 0,
            error: MongoNetworkError: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:513:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:642:26)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at emitErrorNT (node:internal/streams/destroy:157:8)
                at emitErrorCloseNT (node:internal/streams/destroy:122:3)
                at processTicksAndRejections (node:internal/process/task_queues:83:21) {
              cause: Error: getaddrinfo EAI_AGAIN cluster0-shard-00-02.oeari.mongodb.net
                  at GetAddrInfoReqWrap.onlookup [as oncomplete] (node:dns:71:26) {
                errno: -3001,
                code: 'EAI_AGAIN',
                syscall: 'getaddrinfo',
                hostname: 'cluster0-shard-00-02.oeari.mongodb.net'
              },
              [Symbol(errorLabels)]: Set(1) { 'ResetPool' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-13 05:02:11'
}
{
  message: {
    findUserById: MongoNetworkTimeoutError: connection timed out
        at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
        at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
        at Object.onceWrapper (node:events:641:28)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at TLSSocket.Socket._onTimeout (node:net:516:8)
        at listOnTimeout (node:internal/timers:559:17)
        at processTimers (node:internal/timers:502:7) {
      connectionGeneration: 19,
      [Symbol(errorLabels)]: Set(0) {}
    }
  },
  level: 'error',
  timestamp: '2023-06-18 16:19:15'
}
{
  message: {
    findUserById: MongoNetworkTimeoutError: connection timed out
        at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
        at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
        at Object.onceWrapper (node:events:641:28)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at TLSSocket.Socket._onTimeout (node:net:516:8)
        at listOnTimeout (node:internal/timers:559:17)
        at processTimers (node:internal/timers:502:7) {
      connectionGeneration: 19,
      [Symbol(errorLabels)]: Set(0) {}
    }
  },
  level: 'error',
  timestamp: '2023-06-18 16:19:15'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1523691,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:48:44'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:49:49'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:49:50'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:49:50'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:49:50'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:49:50'
}
{
  message: {
    findUserById: MongooseServerSelectionError: Could not connect to any servers in your MongoDB Atlas cluster. One common reason is that you're trying to access the database from an IP that isn't whitelisted. Make sure your current IP address is on your Atlas cluster's IP whitelist: https://www.mongodb.com/docs/atlas/security-whitelist/
        at Function.Model.$wrapCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/model.js:5372:32)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4950:21
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:41:5
        at new Promise (<anonymous>)
        at promiseOrCallback (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/helpers/promiseOrCallback.js:40:10)
        at model.Query.exec (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:4949:10)
        at model.Query.Query.then (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/lib/query.js:5032:15) {
      reason: TopologyDescription {
        type: 'ReplicaSetNoPrimary',
        servers: Map(3) {
          'cluster0-shard-00-00.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-00.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-01.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-01.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          },
          'cluster0-shard-00-02.oeari.mongodb.net:27017' => ServerDescription {
            address: 'cluster0-shard-00-02.oeari.mongodb.net:27017',
            type: 'Unknown',
            hosts: [],
            passives: [],
            arbiters: [],
            tags: {},
            minWireVersion: 0,
            maxWireVersion: 0,
            roundTripTime: -1,
            lastUpdateTime: 1589109,
            lastWriteDate: 0,
            error: MongoNetworkTimeoutError: connection timed out
                at connectionFailureError (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:515:14)
                at TLSSocket.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/mongoose/node_modules/mongodb/src/cmap/connect.ts:415:16)
                at Object.onceWrapper (node:events:641:28)
                at TLSSocket.emit (node:events:527:28)
                at TLSSocket.emit (node:domain:475:12)
                at TLSSocket.Socket._onTimeout (node:net:516:8)
                at listOnTimeout (node:internal/timers:559:17)
                at processTimers (node:internal/timers:502:7) {
              [Symbol(errorLabels)]: Set(2) { 'ResetPool', 'InterruptInUseConnections' }
            },
            topologyVersion: null,
            setName: null,
            setVersion: null,
            electionId: null,
            logicalSessionTimeoutMinutes: null,
            primary: null,
            me: null,
            '$clusterTime': null
          }
        },
        stale: false,
        compatible: true,
        heartbeatFrequencyMS: 10000,
        localThresholdMS: 15,
        setName: 'atlas-80ye5b-shard-0',
        maxElectionId: new ObjectId("7fffffff0000000000000007"),
        maxSetVersion: 1,
        commonWireVersion: 0,
        logicalSessionTimeoutMinutes: null
      },
      code: undefined
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:49:50'
}
{
  message: {
    getTrelloBoardError: FetchError: request to https://api.trello.com/1/organizations/ttpbeta/boards?fields=id,name&key=0d05ca858d1bc7893a91aef2712e7568&token=2d8a9892820b0ddb4d67225b6ab7f4130ccd31fffc6860c3f743e3bf6147af68 failed, reason: getaddrinfo EAI_AGAIN api.trello.com
        at ClientRequest.<anonymous> (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/node_modules/node-fetch/lib/index.js:1505:11)
        at ClientRequest.emit (node:events:527:28)
        at ClientRequest.emit (node:domain:475:12)
        at TLSSocket.socketErrorListener (node:_http_client:454:9)
        at TLSSocket.emit (node:events:527:28)
        at TLSSocket.emit (node:domain:475:12)
        at emitErrorNT (node:internal/streams/destroy:157:8)
        at emitErrorCloseNT (node:internal/streams/destroy:122:3)
        at processTicksAndRejections (node:internal/process/task_queues:83:21) {
      type: 'system',
      errno: 'EAI_AGAIN',
      code: 'EAI_AGAIN'
    }
  },
  level: 'error',
  timestamp: '2023-06-25 12:50:48'
}
{
  message: {
    _updateDepartmentsPriority: TypeError: Cannot read properties of undefined (reading 'map')
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:126:27
        at Generator.next (<anonymous>)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:4:12)
        at _updateDepartmentsPriority (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:152:16)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/presentation/department/department.ts:10:57
        at Generator.next (<anonymous>)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/presentation/department/department.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error',
  timestamp: '2023-07-10 11:40:36'
}
{
  message: {
    _updateDepartmentsPriority: TypeError: Cannot read properties of undefined (reading 'map')
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:126:27
        at Generator.next (<anonymous>)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:8:71
        at new Promise (<anonymous>)
        at __awaiter (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:4:12)
        at _updateDepartmentsPriority (/home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/controllers/department.ts:152:16)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/presentation/department/department.ts:11:58
        at Generator.next (<anonymous>)
        at /home/ezat-elzalouy/Desktop/Unicode/TTP/TTP-Back/src/lib/presentation/department/department.ts:8:71
        at new Promise (<anonymous>)
  },
  level: 'error',
  timestamp: '2023-07-10 11:47:42'
}
